level.go

```diff
@@ -49,18 +52,40 @@ func (f *LevelFilter) Check(line []byte) bool {
 	return !ok
 }
 
+// Write is a specialized implementation of io.Writer suitable for being
+// the output of a logger from the "log" package.
+//
+// This Writer implementation assumes that it will only recieve byte slices
+// containing one or more entire lines of log output, each one terminated by
+// a newline. This is compatible with the behavior of the "log" package
+// directly, and is also tolerant of intermediaries that might buffer multiple
+// separate writes together, as long as no individual log line is ever
+// split into multiple slices.
+//
+// Behavior is undefined if any log line is split across multiple writes or
+// written without a trailing '\n' delimiter.
 func (f *LevelFilter) Write(p []byte) (n int, err error) {
-	// Note in general that io.Writer can receive any byte sequence
-	// to write, but the "log" package always guarantees that we only
-	// get a single line. We use that as a slight optimization within
-	// this method, assuming we're dealing with a single, complete line
-	// of log data.
-
-	if !f.Check(p) {
-		return len(p), nil
+	for len(p) > 0 {
+		// Split at the first \n, inclusive
+		idx := bytes.IndexByte(p, '\n')
```

Comment:
Log message as accepted by the this and the standard logger often have multi-line message. In terraform for example, splitting and filtering on `\n` is going to let through a whole bunch of text graphs with no descriptions. Providers also heavily use multi-line output, though that is going through hclog, so it may already have every line prefixed. In order to filter core logs, I think we're going to have split only on `\n[LEVEL]` where `LEVEL` is only one of those specifically accepted by this logger.

Another issue I haven't dug into is that splitting like this is causing terraform to hang completely. I believe this is due to the read-write loops in prefixedio, but that is just a guess at this point.

Reply:
Yikes, that's right. I forgot that's a thing (newlines in the log messages).